File: auth-jwt/auth-jwt-scala-sbt/auth-jwt-scala-sbt-impl/src/main/scala/com/example/authjwt/impl/AuthJwtServiceImpl.scala
package com.example.authjwt.impl

import akka.NotUsed
import com.example.authjwt.api.AuthJwtService
import com.lightbend.lagom.scaladsl.api.ServiceCall
import com.lightbend.lagom.scaladsl.server.ServerServiceCall
import org.pac4j.core.authorization.authorizer.RequireAnyRoleAuthorizer.requireAnyRole
import org.pac4j.core.config.Config
import org.pac4j.core.profile.CommonProfile
import org.pac4j.lagom.scaladsl.SecuredService

import scala.concurrent.Future

/**
  * Implementation of the AuthJwtService.
  */
class AuthJwtServiceImpl(override val securityConfig: Config) extends AuthJwtService with SecuredService {
  override def headerJwtAuthenticate: ServiceCall[NotUsed, String] = {
    authenticate { profile =>
      // `authenticate` requires a `ServerServiceCall` to access the headers.
      ServerServiceCall { _ =>
        Future.successful(profile.getId)
      }
    }
  }

  override def headerJwtAuthorize: ServiceCall[NotUsed, String] = {
    authorize(requireAnyRole[CommonProfile]("manager"), (profile: CommonProfile) =>
      // `authorize` requires a `ServerServiceCall` to access the headers.
      ServerServiceCall { _: NotUsed =>
        Future.successful(profile.getId)
    })
  }
}

File: auth-jwt/auth-jwt-java-sbt/auth-jwt-java-sbt-impl/src/main/java/com/example/authjwt/impl/AuthJwtServiceImpl.java
package com.example.authjwt.impl;

import akka.NotUsed;
import com.example.authjwt.api.AuthJwtService;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import org.pac4j.core.config.Config;
import org.pac4j.lagom.javadsl.SecuredService;

import javax.inject.Inject;

import static org.pac4j.core.authorization.authorizer.RequireAnyRoleAuthorizer.requireAnyRole;

import static java.util.concurrent.CompletableFuture.completedFuture;

/**
 * Implementation of the AuthJwtService.
 */
public class AuthJwtServiceImpl implements AuthJwtService, SecuredService {

    private final Config securityConfig;

    @Inject
    public AuthJwtServiceImpl(Config securityConfig) {
        this.securityConfig = securityConfig;
    }

    @Override
    public Config getSecurityConfig() {
        return securityConfig;
    }

    @Override
    public ServiceCall<NotUsed, String> headerJwtAuthenticate() {
        return authenticate(profile ->
                request -> completedFuture(profile.getId())
        );
    }

    @Override
    public ServiceCall<NotUsed, String> headerJwtAuthorize() {
        return authorize(requireAnyRole("manager"), profile ->
                request -> completedFuture(profile.getId())
        );
    }
}
